File: backend/src/domain/entities/ForumTopic.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('forum_topics')
export class ForumTopic {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ name: 'created_by_member_id' })
  createdByMemberId: number;
}

File: backend/src/main/java/com/service/backend/forum/entity/ForumTopic.java
package com.service.backend.forum.entity;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table("forum_topics")
public class ForumTopic {
    
    @Id
    private Integer id;
    
    @Column("organization_id")
    private Integer organizationId;
    
    @Column("title")
    private String title;
    
    @Column("created_by_member_id")
    private Integer createdByMemberId;
    
    @Column("category_id")
    private Integer categoryId;
    
    @Column("view_count")
    private Integer viewCount;

    @Column("is_locked")
    private Boolean isLocked;
    
    @Column("created_at")
    private LocalDateTime createdAt;
    
    @Column("updated_at")
    private LocalDateTime updatedAt;
}
